tinytodo/src/context.rs

```diff
@@ -296,26 +294,26 @@ impl AppContext {
     }
 
     fn add_share(&mut self, r: AddShare) -> Result<AppResponse> {
-        self.is_authorized(&r.uid, &*ACTION_EDIT_SHARE, &r.list)?;
-        let list = self.entities.get_list(&r.list)?;
+        let proof = self.is_authorized::<actions::EditShare>(&r.uid, &r.list)?;
+        let list = self.entities.get_list(&r.list, &proof)?;
```

Comment:
nit: Is this different than the witness design pattern? https://willcrichton.net/rust-api-type-patterns/witnesses.html If not, I'd suggest using `witness` as the parameter name.